tools/Unix.mk

```diff
@@ -19,6 +19,14 @@
 ############################################################################
 
 export TOPDIR := ${shell echo $(CURDIR) | sed -e 's/ /\\ /g'}
+WSDIR := ${shell cd "${TOPDIR}"/.. && pwd -P}
+
+export NXTMPDIR := $(WSDIR)/nxtmpdir
+$(info $$NXTMPDIR = $(NXTMPDIR))
+
+ifeq ($(wildcard $(NXTMPDIR)),)
+$(shell mkdir -p $(NXTMPDIR))
```

Comment:
but, it isn't good to always create a temp folder if esp isn't enable in defconfig.
BTW, it should be contained in ci specific script, Unix.mk it's too general.

Reply:
Hi @xiaoxiang781216  
my idea to speed up build is not only about esp but in general all parts where git and curl is used even to apps (draft for lvgl https://github.com/apache/nuttx-apps/commit/cbc3cc42985efcfcc2e999500498f557f282c9e3) when possible. 

.github/workflows/build.yml:
added option -S  adds the nxtmpdir folder for third-party packages.
./cibuild.sh -c -A -N -R -S testlist/${{matrix.boards}}.dat